youtube_dl/downloader/fragment.py

```diff
@@ -116,6 +116,7 @@ def _append_fragment(self, ctx, frag_content):
         finally:
             if self.__do_ytdl_file(ctx):
                 self._write_ytdl_file(ctx)
+            self.mtime = os.stat(ctx['fragment_filename_sanitized']).st_mtime
```

Comment:
Nothing conceptually changed: you still request `st_mtime` unnecessarily *n-1* times.